mail_message.go

```diff
@@ -351,6 +354,39 @@ func (message *MailMessage) postToBlog(senderUser *User) error {
 	return nil
 }
 
+// Sends email data to a Mattermost webhook to post on the town-square channel
+func (message *MailMessage) postToMattermost() {
+	body := fmt.Sprintf("@channel: %s", message.body.HTML)
+
+	data := struct {
+		Channel  string `json:"channel"`
+		Username string `json:"username"`
+		Icon_url string `json:"icon_url"`
+		Text     string `json:"text"`
+	}{config.GetString("mattermost_channel_name"), config.GetString("mattermost_bot_username"), config.GetString("mattermost_icon_url"), body}
+
+    jsonData, err := json.Marshal(data)
+    if err != nil {
+        fmt.Println(err)
+        return
+    }
+	
+	payload := fmt.Sprintf("'%s'", string(jsonData))
```

Comment:
Is this really necessary? I would think that you can pass `string(jsonData)` directly as the request body.